Sample File Updates

etimberg opened the pull request:
I added a new helper, `samples/chartColors.js` that defines the colors that match the docs. I updated all of the samples to use these colours.

I improved sample organization, with common samples grouped into folders. I added new samples that allow comparing different settings. For example, `samples/tooltips/interaction-modes.html` creates a number of charts that allow the user to compare the different interaction mode settings.

simonbrunel commented on samples/chartColors.js:
```diff
@@ -0,0 +1,9 @@
+window.chartColors = {
```
We could generalize this file and move in it all samples helpers (e.g. `randomScalingFactor`)? It would be renamed to `helpers.js` or `utils.js`? Similar to the [utils.js file](/chartjs/chartjs-plugin-deferred/tree/master/samples) in the deferred plugin.

etimberg replied:
Sure, I can do that. It will definitely simplify things.

etimberg commented:
I will rename the animation folder. I like the idea of putting the scales under `scales/type/*` so I will make that change as well

etimberg commented:
Renamed `chartColors.js` to `utils.js`. I moved `randomScalingFactor` into there. I also moved the files around so that scales are in a better spot and `AnimationCallbacks` is now `animation`.
